### PR TITLE
Story/rhoaieng 2983 experiments data filters

### PR DESCRIPTION
**Pull request by Gkrumbach07**
closes: https://issues.redhat.com/browse/RHOAIENG-2983

## Description

- added tables and data fetching for archived and active experiments
- added filters

Screenshot 2024-02-29 at 3 22 15 PM
Screenshot 2024-02-29 at 3 22 05 PM

## How Has This Been Tested?

- create some experiments and archive some in the KF UI
- see them in dashboard UI
- try the filters and sorting

## Test Impact

added filter tests and data vis tests

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit tests & storybook for related changes)

If you have UI changes: 

- [x] Included any necessary screenshots or gifs if it was a UI change.
- [x] Included tags to the UX team if it was a UI/UX change (find relevant UX in the [SMEs](https://github.com/opendatahub-io/odh-dashboard/tree/main/docs/smes.md) section).

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

**Comment by Gkrumbach07** [comment]
/hold
This cherry picks the commits off of https://github.com/opendatahub-io/odh-dashboard/pull/2514

wait until this merges to merge this

**Comment by Gkrumbach07** [comment]
cc @yannnz @xianli123 @kywalker-rh

This is the experiments table. some notes
- the status icons have a tooltip that is the same as the status icon in the runs table (Failed, Succeeded, ...)
- not really a good way to sort on last 5 runs so did not add that
- Experiment name will have a link but that comes in a later Jira

**Comment by Gkrumbach07** [comment]
PR is ready for re-review. 

UX was addressed except for bulk select. This is not possible within the current KF api due to pagination limitations. Also, PF seems to not support some of the checkbox UX despite one of their example demos showing it

**Review by andrewballantyne — COMMENTED** [review]
Something for us to work on; code looks good -- let me run it and play around with the UI.

**Comment by andrewballantyne** [comment]
I think we want to go with the same design pattern as the Runs page -- where the tabs touch the side. 
image

I think this is simply by just setting `provideChildrenPadding=false` on the `ApplicationPage`

File: frontend/src/api/pipelines/custom.ts
Comment (Gkrumbach07) [reply frontend/src/api/pipelines/custom.ts:126]: yeah i agree. it was a last min change needed so ill think it through more on the following experiments prs

**Review by andrewballantyne — APPROVED** [review]

**Pull request merged**
